clearml_agent/helper/repo.py

```diff
@@ -253,6 +254,16 @@ def get_username(user_, password=None):
         match = cls.SSH_URL_GIT_SYNTAX.match(url)
         if match:
             user, host, path = match.groups()
+
+            # handle special azure cases
+            if "ssh" and "azure" in host:
```

Comment:
This seems pretty specific and I'm not sure we can count on it not appearing in other URLs - I think we need a better way to detect that

Reply:
For sure. I'm just not aware of other URLs that has this pattern. 

Generally I guess that you wouldn't expect `[.?]ssh[.?]` to show up in the host path in any case and then we only need to insert the /_git/ thing for azure URLS?

Reply:
We could also add a match in the `SSH_URL_GIT_SYNTAX` regex that filters out `ssh.` from the path and then only handle the special `_git` component here.

Reply:
WDYT @jkhenning?

Reply:
To make it more general, I suppose one could filter for `v` followed by a single digit, instead of `v3` directly, to mitigate v4 and v5 etc. It does feel over-engineered though.

Reply:
@mads-oestergaard I think we should just go with an exact string representing the Azure URL (this can be in the default configuration so people might override it if things change)